h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -65,23 +65,28 @@
  * Model[] models = grid.getModels()
  * }</pre>
  *
- * @see hex.grid.HyperSpaceWalker
+ * @see HyperSpaceWalker
  * @see #startGridSearch(Key, HyperSpaceWalker)
  */
 public final class GridSearch<MP extends Model.Parameters> extends Keyed<GridSearch> {
   public final Key<Grid> _result;
   public final Job<Grid> _job;
+  public final int _parallelismLevel;
+
+  public static final int SEQUENTIAL_MODE = 1;
```

Comment:
why not enum? I think we should have 2 arguments here instead of ints with hidden meaning - paralelism type (signle, parallel_adaptive, parallel_fixed) and an int argument for parallel_fixed case, better still dont use enum but use an object that contains all the info

Reply:
Ints are dangerous and if you have a look at the commits, it used to have two arguments:
- Parallelism enabled/disabled
- Degree of parallelism

Then, I discussed this with more customer-facing people in the office and they all suggested libraries like scikit have integers only and this is more understandable. Hence my motivation to use integer.

Not sure which way is better. Personally, I like two params more.